contest.hpp

```diff
@@ -143,35 +149,50 @@ export class Contest {
         winner = Role {};
         should_giveup = false;
     }
-
+    Contest(){
+        clear();
+    }
     void reject()
     {
-        if (status != Status::NOT_PREPARED)
+        if (status != Status::NOT_PREPARED){
+            logger->error("status: {}",(int)status);
```

Comment:
Why two lines? BTW what's the difference between `error` & `critical`?

Reply:
Critical will cause system falling down, but error won't. In my opinion, the exception should be catch and solve in the future.